source/common/common/logger.cc

```diff
@@ -240,31 +240,35 @@ std::vector<Logger>& Registry::allLoggers() {
   return *all_loggers;
 }
 
-spdlog::logger& Registry::getLog(Id id) { return *allLoggers()[static_cast<int>(id)].logger_; }
+spdlog::logger& Registry::getLog(Id id) { return *allLoggers()[static_cast<int>(id)].getLogger(); }
```

Comment:
With the old interface we didn't create a temporary shared pointer (e.g. atomic ref count increment), but now we do :( and we end up throwing it away. Can we implement this without adding this overhead?

Reply:
Good point. I am trying to return the reference itself, can you take a look now? Thank you!